internal/controller/pipelinerollout_controller.go

```diff
@@ -685,23 +690,6 @@ func (r *PipelineRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return nil
 }
 
-// pipelineSpecNeedsUpdating() tests for essential equality, with any irrelevant fields eliminated from the comparison
-func pipelineSpecNeedsUpdating(ctx context.Context, a *kubernetes.GenericObject, b *kubernetes.GenericObject) (bool, error) {
-	numaLogger := logger.FromContext(ctx)
-	// remove lifecycle.desiredPhase field from comparison to test for equality
-	pipelineWithoutDesiredPhaseA, err := pipelineWithoutDesiredPhase(a)
-	if err != nil {
-		return false, err
-	}
-	pipelineWithoutDesiredPhaseB, err := pipelineWithoutDesiredPhase(b)
-	if err != nil {
-		return false, err
-	}
-	numaLogger.Debugf("comparing specs: pipelineWithoutDesiredPhaseA=%v, pipelineWithoutDesiredPhaseB=%v\n", pipelineWithoutDesiredPhaseA, pipelineWithoutDesiredPhaseB)
-
-	return !reflect.DeepEqual(pipelineWithoutDesiredPhaseA, pipelineWithoutDesiredPhaseB), nil
-}
```

Comment:
replaced this with `childNeedsUpdating()` which is one of the functions of the `rolloutObject` interface